### PR TITLE
[Compute] Disk 2022-03-02

### PR DESCRIPTION
**Pull request by haagha**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/disk.json [hunk 1]
```diff
@@ -0,0 +1,856 @@
...
+          "List all managed disks in a subscription.": {
+            "$ref": "./examples/diskExamples/Disk_ListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}/beginGetAccess": {
```
Comment (jorgecotillo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/disk.json:361]: >beginGetAccess [](http://example.com/codeflow?start=117&length=14)

What does begin - end access does? Are you getting a token? handler? The description is not clear and API kind of explains what it does but doesn't explain access to which entity (e.g. resource group, VM?)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskAccess.json [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskAccesses/{diskAccessName}": {
+      "put": {
```
Comment (jorgecotillo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskAccess.json:38]: >"put" [](http://example.com/codeflow?start=6&length=5)

Make sure to follow (in all your specs) our async pattern: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put) 

e.g. return 201 Created on async PUT

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskEncryptionSet.json [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+          "List all disk encryption sets in a subscription.": {
+            "$ref": "./examples/diskEncryptionSetExamples/DiskEncryptionSet_ListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskEncryptionSets/{diskEncryptionSetName}/associatedResources": {
```
Comment (jorgecotillo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskEncryptionSet.json:331]: >associatedResources [](http://example.com/codeflow?start=143&length=19)

Are these proxy resources? Seems like it but would like to confirm. Also if you support a collection GET you must support a single GET (see our armwiki: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review) - Inconsistent or incomplete CRUD)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRestorePoint.json [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{vmRestorePointName}/diskRestorePoints/{diskRestorePointName}": {
```
Comment (jorgecotillo) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRestorePoint.json:37]: >diskRestorePoints [](http://example.com/codeflow?start=188&length=17)

How are these created?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json [hunk 5]
```diff
@@ -0,0 +1,593 @@
...
+      "type": "object",
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "description": "Indicates the error code if the background copy of a resource created via the CopyStart operation fails.",
+          "enum": [
+            "CopySourceNotFound"
+          ],
+          "x-ms-enum": {
+            "name": "CopyCompletionError",
```
Comment (ArcturusZhang) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json:428]: We have a model and an enum type with the same name, this is not allowed. Please either rename the model on line 418 or rename the type name of the enumeration on line 428

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json (hunk 5 above)
Comment (kkaran27) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json:428]: Created a PR - https://github.com/Azure/azure-rest-api-specs/pull/19145

Comment (kkaran27) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json:428]: New PR - https://github.com/Azure/azure-rest-api-specs/pull/19160

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
The cross-version breaking change is not working since the structure of the swagger files has been changed

**Pull request merged**